cmd/client-s3.go

```diff
@@ -1699,6 +1699,9 @@ func (c *S3Client) List(ctx context.Context, opts ListOptions) <-chan *ClientCon
 // it falls back to the regular listing if not.
 func (c *S3Client) versionedList(ctx context.Context, contentCh chan *ClientContent, opts ListOptions) {
 	b, o := c.url2BucketAndObject()
+	if opts.IsDir {
+		o += "/"
+	}
```

Comment:
This kind of manual append is problematic @adfost is this not possible for the console to do this as necessary? - since you already know that `IsDir: true`

Reply:
No it isn't possible, regardless of whether it is a directory it treats it like a file. It is necessary for mc to know that it is a directory, not a file.

Reply:
he's right, we don't need to do this, we can simply pass this on the prefix as we access the file